waspc/cli/Wasp/Cli/Command/Watch.hs

```diff
@@ -80,9 +82,11 @@ watch waspProjectDir outDir = FSN.withManager $ \mgr -> do
     recompile :: IO ()
     recompile = do
       waspSays $ asWaspStartMessage "Recompiling on file change..."
-      compilationResult <- compileIO waspProjectDir outDir
+      compilationResult <- runExceptT $ do
+        ExceptT $ compileIO waspProjectDir outDir
+        checkIfDbSchemaMatchesChecksum outDir
```

Comment:
You will see a few instances where I mix `ExceptT String IO ()` and `IO (Either String ())` with a `runExceptT` such that I get short circuiting. Is this a standard way of doing these? I did not in many cases want to for all of them into the `ExceptT` model.

Reply:
There is another use of `checkIfDbSchemaMatchesChecksum` in a `runExceptT` as well.

Reply:
I checked `checkIfDbSchemaMatchesChecksum` and I don't see it benefiting from being `ExceptT`. So let's imagine it is `Either` -> in that case you could refactor this as `compilationResult <- compileIO waspProjectDir outDir >> checkIfDbSchemaMatchesChecksum outDir`, and that should be it if I am correct? So it seems to me there is no reason to use `ExceptT` here.

Reply:
Yeah, I think you are correct. I think I was in gungho monad transformer mode for Prisma errors and tried making everything into an `ExceptT` 😅  Will update thanks 👍🏻

Reply:
Actually, I think we still do need `ExceptT` to short circuit in both cases (Watch.hs and Start.hs). `>>` here is in the context of `IO` still, so it will just execute all of them even if we return a `Left` from any because IO is like the honey badger here, "it don't care". :D

Check this playground out: https://replit.com/@smcnc/ProfuseThistleBlogclient

I can still make them all `IO (Either String ())` again though if that is easier to read and can just use `ExceptT $ ` before each inside a `runExceptT $ do` to get our desired short circuit behavior.

Reply:
There are no longer changes to Watch.hs in the current approach. The check now happens during Generation.